src/screens/MainApp/MyDiagnoses/renderUtilities/index.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import * as StorageService from '~/storageService'
+import Diagnose from '../Diagnose'
+
+const getURL = async (imageReference) => {
+  try {
+    return await StorageService.getDownloadURL(imageReference)
+  } catch (error) {
+    return null
+  }
+}
+
+const renderDiagnose = async (diagnose, key) => {
+  const thumbnail = await getURL(diagnose.imageReferences[0])
+  let answerQuantity = 0
+
+  if (diagnose.answer) {
+    answerQuantity = 1
+  }
+
+  return (
+    <Diagnose
+      key={key}
+      thumbnail={thumbnail}
+      firebaseTimestamp={diagnose.createdAt}
+      description={diagnose.text}
+      answerQuantity={answerQuantity}
+    />
+  )
+}
+
```

Comment:
I don't like this asynchronous rendering. Rendering is meant to be a synchronous process. In the future, React will introduce concurrent rendering. I don't know how well asynchronous rendering will play with concurrent rendering. If you need to perform asynchronous actions as part of the rendering process of a component, you have to do so when React tells you. That's the purpose of `useEffect`.